assets/js/tx-sender.js

```diff
@@ -0,0 +1,465 @@
+    const options = {
+      minLedgerVersion: (await api.getLedger()).ledgerVersion,
+      maxLedgerVersion: prepared.instructions.maxLedgerVersion
+    }
+
+    try {
+      // Sign, submit
+      sign_response = api.sign(prepared.txJSON, use_secret)
+      await api.submit(sign_response.signedTransaction)
+    } catch (error) {
+      console.log(error)
+      if (!silent) {
+        errorNotif("Error signing & submitting "+tx_object.TransactionType+" tx: "+error)
+      }
+      return
+    }
+
+    // Wait for tx to be in a validated ledger or to expire
+    try {
+      const data = await verify_transaction(sign_response.id, options)
+      const final_result = data.outcome.result
+      // Future feature: output should link to a TestNet tx lookup/explainer
+      if (final_result === "tesSUCCESS") {
+        if (!silent) {
+          successNotif(tx_object.TransactionType+" tx succeeded (hash: "+sign_response.id+")")
+        }
+      } else {
+        if (!silent) {
+          errorNotif(tx_object.TransactionType+" tx failed w/ code "+final_result+
+                      " (hash: "+sign_response.id+")")
+        }
+      }
+      update_xrp_balance()
+      return data
+    } catch(error) {
+      console.log(error)
+      if (!silent) {
+        errorNotif("Error submitting "+tx_object.TransactionType+" tx: "+error)
+      }
+    }
+
+  }
+
+  //////////////////////////////////////////////////////////////////////////////
+  // Issuer Setup for Partial Payments
+  // (Partial payments must involve at least one issued currency, so we set up
+  //  an issuer for a fake currency to ripple through.)
+  //////////////////////////////////////////////////////////////////////////////
+
+  let pp_issuer_address
+  let pp_sending_currency = "BAR"
+  async function set_up_for_partial_payments() {
+    while (!connection_ready) {
+      console.debug("... waiting for connection before doing partial payment setup")
+      await timeout(200)
+    }
+    console.debug("Starting partial payment setup...")
+    $("#pp_progress .progress-bar").addClass("progress-bar-animated")
+    // 1. Get a funded address to use as issuer
+    let pp_issuer_secret
+    try {
+      const faucet_response = await ($.ajax({
+        url: FAUCET_URL,
+        type: 'POST',
+        dataType: 'json'
+      }))
+      pp_issuer_address = faucet_response.account.address
+      pp_issuer_secret = faucet_response.account.secret
+    } catch(error) {
+      console.log("Error getting issuer address for partial payments:", error)
+      return
+    }
+    $("#pp_progress .progress-bar").width("20%")
+
+    // 2. Set DefaultRipple on issuer
+    let resp = await submit_and_verify({
+      TransactionType: "AccountSet",
+      Account: pp_issuer_address,
+      SetFlag: 8
+    }, pp_issuer_secret, true)
+    if (resp === undefined) {
+      console.log("Couldn't set DefaultRipple for partial payment issuer")
+      return
+    }
+    $("#pp_progress .progress-bar").width("40%")
```

Comment:
It's a nice idea, but I'd rather keep the thing visually compact rather than adding a checklist. I could potentially have it update the progress bar text as each step is in progress, though.

My goal is to make this compact enough to fit comfortably on a typical smartphone screen so you can use a totally separate device to send the transactions. (There are some other things I need to tweak for that.)